src/behaviour.rs

```diff
@@ -128,116 +137,115 @@ impl PeerBehaviour {
+    pub fn subscribe(&mut self, topics: Vec<Topic>) -> Result<(), gossipsub::SubscriptionError> {
         for topic in topics {
             self.gossip.subscribe(&topic.into())?;
         }
 
         Ok(())
     }
 
-    /// Send [Message] to peer with the given address
-    pub fn send_to(
-        &mut self,
-        address: PublicAddress,
-        msg: Message,
-    ) -> Result<MessageId, PublishError> {
-        let topic = Topic::Mailbox(address).hash();
-        let content: Vec<u8> = msg.into();
-        self.gossip.publish(topic, content)
+    /// Publish a [Message] to peers subscribed to the [Topic]
+    pub fn publish(&mut self, topic: Topic, msg: Message) -> Result<MessageId, PublishError> {
+        self.gossip.publish(topic.hash(), msg)
     }
 
-    /// Broadcast [Message] with a specific topic
-    pub fn broadcast(
-        &mut self,
-        topic: IdentTopic,
-        msg: Message,
-    ) -> Result<MessageId, PublishError> {
-        let content: Vec<u8> = msg.into();
-        self.gossip.publish(topic, content)
-    }
-
-    /// Check if the [gossipsub::Message] is subscribed by this peer
-    pub fn is_subscribed(&self, message: &gossipsub::Message) -> bool {
-        self.gossip.topics().any(|topic| message.topic.eq(topic))
+    /// Check if the [gossipsub::TopicHash] is subscribed by this peer
+    pub fn is_subscribed(&self, topic_hash: &TopicHash) -> bool {
+        self.gossip.topics().any(|topic| topic_hash.eq(topic))
     }
 }
 
-/// The definition of Out-Event required by [PeerBehaviour].
-pub(crate) enum PeerNetworkEvent {
+/// The definition of Out-Event required by [Behaviour].
+pub(crate) enum NetworkEvent {
     Kad(KademliaEvent),
     Gossip(gossipsub::Event),
     Ping(ping::Event),
     Identify(identify::Event),
 }
 
-impl From<gossipsub::Event> for PeerNetworkEvent {
+impl From<gossipsub::Event> for NetworkEvent {
     fn from(event: gossipsub::Event) -> Self {
         Self::Gossip(event)
     }
 }
 
-impl From<KademliaEvent> for PeerNetworkEvent {
+impl From<KademliaEvent> for NetworkEvent {
     fn from(event: KademliaEvent) -> Self {
         Self::Kad(event)
     }
 }
 
-impl From<ping::Event> for PeerNetworkEvent {
+impl From<ping::Event> for NetworkEvent {
     fn from(event: ping::Event) -> Self {
         Self::Ping(event)
     }
 }
 
-impl From<identify::Event> for PeerNetworkEvent {
+impl From<identify::Event> for NetworkEvent {
     fn from(event: identify::Event) -> Self {
         Self::Identify(event)
     }
 }
 
 #[cfg(test)]
-
 mod test {
     use std::net::Ipv4Addr;
 
-    use super::PeerBehaviour;
-    use crate::{Config, conversions, messages::{Topic, MessageTopicHash}, constants};
-
     use libp2p::{
         gossipsub,
-        Multiaddr, PeerId,
+        identity::{PeerId, ed25519},
+        Multiaddr,
     };
-    use pchain_types::cryptography::PublicAddress;
+    use pchain_types::cryptography::{PublicAddress, self};
```

Comment:
just import `Keypair` here instead of doing `self` here and qualify it as `crytography::Keypair`